sequence_processing_pipeline/Pipeline.py

```diff
@@ -249,23 +249,29 @@ def _validate_sample_sheet(self, sample_sheet_path):
         raise PipelineError('Sample-sheet has the following errors:\n'
                             '\n'.join([str(x) for x in msgs]))
 
-    def generate_sample_information_files(self):
+    def generate_sample_info_files(self, addl_info=None):
         """
         Generate sample-information files in self.output_path.
+        :param addl_info: A df of (sample-name, project-name) pairs.
         :return: A list of paths to sample-information-files.
         """
         if self.mapping_file is not None:
+            # Generate a list of BLANKs for each project.
             df = self.mapping_file[['sample_name', 'project_name']]
-            samples = list(df.to_records(index=False))
-            samples = [x for x in samples if x[0].startswith('BLANK')]
-            projects = list(set([y for x, y in samples]))
         else:
-            samples = []
-            for sample in self.sample_sheet.samples:
-                if sample['Sample_ID'].startswith('BLANK'):
-                    samples.append((sample['Sample_ID'],
-                                    sample['Sample_Project']))
-            projects = list(set([y for x, y in samples]))
+            # Aggregate all data into a DataFrame
+            data = [[x['Sample_ID'], x['Sample_Project']] for
+                    x in self.sample_sheet.samples]
+            df = pd.DataFrame(data, columns=['sample_name', 'project_name'])
+
+        if addl_info is not None:
+            df = pd.concat([df, addl_info],
+                           ignore_index=True).drop_duplicates()
```

Comment:
Do we really need to remove duplicates?

Reply:
We do. df ostensibly holds some or all of the BLANKs we expect. addl_info holds all BLANKs as pulled from Qiita. Unless we decide to drop the contents of df and just use what comes out of Qiita instead of the prep-files, there will be duplicates we need to remove before writing out the SIFs.